### PR TITLE
net: lwm2m: update internal API select_writer()

### PR DESCRIPTION
**Pull request by rbtchc**
As per review of PR #5893, this is a follow-up patch to update
select_writer() API to match the behavior of select_reader() API.

Quote from OMA-TS-LightweightM2M-V1_0_1-20170704-A. 8.2.5
"An Object Instance or Resource is Read by sending a CoAP GET to the
 corresponding path. The response includes the value in the
 corresponding Plain Text, Opaque, TLV or JSON format according to
 the specified Content-Format (see section 6.4).The request MAY
 specify an Accept option containing the preferred Content-Format
 to receive. When the specified Content-Format is not supported by
 the LwM2M Client, the request MUST be rejected."

Therefore, we do not attempt to assign a content-format when the
requested one is not supported. Instead, we return an error code.

Signed-off-by: Robert Chou <robert.ch.chou@acer.com>

### CODE REVIEW

**Review by mike-scott — CHANGES_REQUESTED** [review]
Couple of nits, but patch overall looks good.

File: subsys/net/lib/lwm2m/lwm2m_engine.c [hunk 1]
```diff
@@ -1190,7 +1186,7 @@ static int select_reader(struct lwm2m_input_context *in, u16_t format)
 		return -ENOMSG;
 	}
 
-	return format;
+	return 0;
```
[the remark this answers is not in the document]
Comment (rbtchc) [reply subsys/net/lib/lwm2m/lwm2m_engine.c]: Update commit subject to "net: lwm2m: update internal API select_writer() and select_reader()"

And add description below to the commit message
> 0 is returned when reader or writer has been successfully selected by select_reader() or select_writer()

File: subsys/net/lib/lwm2m/lwm2m_engine.c [hunk 2]
```diff
@@ -1159,15 +1159,11 @@ static u16_t select_writer(struct lwm2m_output_context *out, u16_t accept)
 #endif
 
 	default:
-		SYS_LOG_ERR("Unknown Accept type %u, using LWM2M plain text",
```
[the remark this answers is not in the document]
Comment (rbtchc) [reply subsys/net/lib/lwm2m/lwm2m_engine.c]: added back the warning message

**Review by mike-scott — APPROVED** [review]
LGTM

**Comment by rbtchc** [comment]
@jukkar please help to review the PR. Thanks.

**Review by nashif — APPROVED** [review]

**Pull request merged**